test/e2e_node/split_disk_test.go

```diff
@@ -0,0 +1,315 @@
+		// Therefore, Node.Fs should be the same as Runtime.ContainerFs.
+		gomega.Expect(summary.Node.Fs.AvailableBytes).To(gomega.Equal(summary.Node.Runtime.ContainerFs.AvailableBytes))
+		gomega.Expect(summary.Node.Fs.CapacityBytes).To(gomega.Equal(summary.Node.Runtime.ContainerFs.CapacityBytes))
+	})
+
+	f.Context("when there is disk pressure", framework.WithSlow(), framework.WithSerial(), framework.WithDisruptive(), func() {
+		f.Context("on imageFs", func() {
+			tempSetCurrentKubeletConfig(f, func(ctx context.Context, initialConfig *kubeletconfig.KubeletConfiguration) {
+				initialConfig.EvictionHard = map[string]string{
+					string(evictionapi.SignalNodeFsAvailable):      "30%",
+					string(evictionapi.SignalContainerFsAvailable): "30%",
+					string(evictionapi.SignalImageFsAvailable):     "30%",
+				}
+				initialConfig.EvictionMinimumReclaim = map[string]string{}
+				ginkgo.By(fmt.Sprintf("EvictionHard %s", initialConfig.EvictionHard))
+			})
+
+			runImageFsPressureTest(f, pressureTimeout, expectedNodeCondition, logDiskMetrics, []podEvictSpec{
+				{
+					evictionPriority: 1,
+					pod:              innocentPod(),
+				},
+			})
+		})
+
+		f.Context("on containerFs", func() {
+			expectedStarvedResource := v1.ResourceEphemeralStorage
+			diskTestInMb := 5000
+
+			tempSetCurrentKubeletConfig(f, func(ctx context.Context, initialConfig *kubeletconfig.KubeletConfiguration) {
+				initialConfig.EvictionHard = map[string]string{
+					string(evictionapi.SignalNodeFsAvailable):  "30%",
+					string(evictionapi.SignalImageFsAvailable): "30%",
+				}
+				initialConfig.EvictionMinimumReclaim = map[string]string{}
+				ginkgo.By(fmt.Sprintf("EvictionHard %s", initialConfig.EvictionHard))
+			})
+			runEvictionTest(f, pressureTimeout, expectedNodeCondition, expectedStarvedResource, logDiskMetrics, []podEvictSpec{
+				{
+					evictionPriority: 1, // This pod should exceed disk capacity on nodeFs since writing to writeable layer
+					pod: diskConsumingPod("container-emptydir-disk-limit", diskTestInMb, nil,
+						v1.ResourceRequirements{}),
+				},
+			})
+		})
+	})
+})
+
+// runImageFsPressureTest tests that
+// Similar to eviction test but skipping the checks on eviction
+// Want to induce disk pressure on a imageFs
+func runImageFsPressureTest(f *framework.Framework, pressureTimeout time.Duration, expectedNodeCondition v1.NodeConditionType, logFunc func(ctx context.Context), testSpecs []podEvictSpec) {
+	// Place the remainder of the test within a context so that the kubelet config is set before and after the test.
+	ginkgo.Context("", func() {
+		ginkgo.BeforeEach(func(ctx context.Context) {
+			// Reduce memory usage in the allocatable cgroup to ensure we do not have MemoryPressure
+			reduceAllocatableMemoryUsageIfCgroupv1()
+			// Nodes do not immediately report local storage capacity,
+			// so sleep a little to allow pods requesting local storage not to fail to be scheduled.
```

Comment:
```suggestion
			// Nodes do not immediately report local storage capacity,  so sleep a little
			// to allow pods requesting local storage not to fail to be scheduled.
```

Reply:
I prefer to split a sentence at comma if there is no significant difference in the length.
It's more readable to me.

Reply:
As long as you will keep line lengths aligned, I am fine with it. Otherwise, like the second like, things look awkward and stick out. 🙂

Reply:
I made it simple. please check it.

Reply:
@bitoku, looks good! :+1: Thank you!